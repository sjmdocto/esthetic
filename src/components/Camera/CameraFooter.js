import React from 'react';
import {View} from 'react-native';
import Icon from 'react-native-vector-icons/Feather';
import {styles} from './CameraFooter.style';
import theme from '../../styles/theme.style';

/**
 * @function CameraFooter
 * @param {*} props
 * @param {function} props.closeCamera
 * @param {function} props.takePicture
 */
const CameraFooter = (props) => {
  return (
    <View style={styles.footerContainer}>
      <View style={styles.backButton}>
        <Icon
          name="arrow-left"
          size={25}
          color={theme.EST_ORANGE}
          onPress={props.closeCamera}
          accessible={true}
          accessibilityLabel={'Back button'}
        />
      </View>
      <View style={styles.shutterButton}>
        <Icon
          name="aperture"
          size={35}
          color={'white'}
          onPress={props.takePicture}
          accessible={true}
          accessibilityLabel={'Shutter button'}
        />
      </View>
    </View>
  );
};

export default CameraFooter;
