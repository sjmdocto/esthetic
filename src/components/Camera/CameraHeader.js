import React from 'react';
import {SafeAreaView, Text} from 'react-native';
import {styles} from './Camera.style';

const CameraHeader = () => {
  return (
    <SafeAreaView style={styles.topContainer}>
      <Text style={styles.title}>Add New Clothes</Text>
    </SafeAreaView>
  );
};

export default CameraHeader;
