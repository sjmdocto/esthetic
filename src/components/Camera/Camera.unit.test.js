import React from 'react';
import CameraHeader from './CameraHeader';
import CameraFooter from './CameraFooter';
import {render, fireEvent} from 'react-native-testing-library';

describe('Camera test suite:', () => {
  const mockBackButton = jest.fn().mockName('closeCamera');
  const mockTakePicture = jest.fn().mockName('takePicture');

  test('Header renders properly', () => {
    const {toJSON} = render(<CameraHeader />);
    expect(toJSON()).toMatchSnapshot();
  });

  test('Footer renders properly', () => {
    const {toJSON} = render(<CameraFooter />);
    expect(toJSON()).toMatchSnapshot();
  });

  test('Back button is functional', () => {
    const {getByLabelText} = render(
      <CameraFooter closeCamera={mockBackButton} />,
    );
    fireEvent.press(getByLabelText('Back button'));
    expect(mockBackButton).toHaveBeenCalled();
  });

  test('Save button is functional', () => {
    const {getByLabelText} = render(
      <CameraFooter takePicture={mockTakePicture} />,
    );
    fireEvent.press(getByLabelText('Shutter button'));
    expect(mockTakePicture).toHaveBeenCalled();
  });
});
