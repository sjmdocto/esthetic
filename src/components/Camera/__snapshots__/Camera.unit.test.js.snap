// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Camera test suite: Footer renders properly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "black",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "left": 125,
        "position": "absolute",
      }
    }
  >
    <Text
      accessibilityLabel="Back button"
      accessible={true}
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "rgb(227, 131, 4)",
            "fontSize": 25,
          },
          undefined,
          Object {
            "fontFamily": "Feather",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "position": "absolute",
      }
    }
  >
    <Text
      accessibilityLabel="Shutter button"
      accessible={true}
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 35,
          },
          undefined,
          Object {
            "fontFamily": "Feather",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`Camera test suite: Header renders properly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#E5E5E5",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 17,
      }
    }
  >
    Add New Clothes
  </Text>
</RCTSafeAreaView>
`;
